pkg/spanstat/spanstat.go

```diff
@@ -32,7 +34,13 @@ func (s *SpanStat) Start() {
 
 // End ends the current span and adds the measured duration to the total
 func (s *SpanStat) End() {
-	s.totalDuration += time.Since(s.spanStart)
+	if !s.spanStart.IsZero() {
+		s.totalDuration += time.Since(s.spanStart)
+	} else {
+		stackdump.Errorf("End() called without prior Start()")
```

Comment:
I think the only plausible justification for a panic is if the system is inconsistent and continued operation is not possible. If we panic because of harmless errors like these, we are just causing downtime in systems for no good reason. I'd rather scan logs for these errors than have users yell at us because we panic for a minor bug like this.

Reply:
But we should caught the panics in our CI. Also, golang itself has lots of panics built-in for situations like these. For example, if you call a "WaitGroup.Add()" after a `WaitGroup.Done()` you get a panic. https://play.golang.org/p/5OeXcShdnev

Reply:
An `.Add()` after a `.Done()` can potentially cause a deadlock and may justify this. At the same time, the golang runtime does not have a standardized way of reporting such problems without introducing error handling into every API function.

I really don't buy into the value of crashing the agent and cause a disruption in all L7 forwarding for simple developer mistakes. I'm happy to be convinced otherwise but we need more justification than just "others are doing similar".

Reply:
@raybejjani I like that idea a lot! Will add this.

Reply:
Complication. This will now no longer allow to actually test the spanstat package to catch these conditions without causing panics.

Reply:
I guess it is possible to to keep the build tag out of the `go test` invocation and only pass it into the build.

Reply:
If you add the string to https://github.com/cilium/cilium/blob/master/test/helpers/cons.go#L192-L203 the CI will also catch this (not quite the panic aanm wants but it is still developer feedback).